src/content-helper/editor-sidebar/smart-linking/component.tsx

```diff
@@ -253,49 +270,88 @@ export const SmartLinkingPanel = ( {
 		} catch ( e: any ) { // eslint-disable-line @typescript-eslint/no-explicit-any
-			setError( e );
-			// eslint-disable-next-line no-console
-			console.error( e );
-			createNotice( 'error', __( 'There was a problem applying smart links.', 'wp-parsely' ), {
+			let snackBarMessage = __( 'There was a problem applying smart links.', 'wp-parsely' );
+
+			// Handle the case where the operation was aborted by the user.
+			if ( e.code && e.code === ContentHelperErrorCode.ParselyAborted ) {
+				snackBarMessage = sprintf(
+					/* translators: %d: number of retry attempts, %s: attempt plural */
+					__( 'The Smart Linking process was cancelled after %1$d %2$s.', 'wp-parsely' ),
+					e.numRetries,
+					_n( 'attempt', 'attempts', e.numRetries, 'wp-parsely' )
+				);
+				e.message = snackBarMessage;
+			}
+
+			await setError( e );
+			createNotice( 'error', snackBarMessage, {
 				type: 'snackbar',
 				isDismissible: true,
 			} );
 		} finally {
 			await setLoading( false );
 			await setApplyTo( previousApplyTo );
+			await setIsRetrying( false );
 			await removeOverlay( isFullContent ? 'all' : selectedBlock?.clientId );
 			clearTimeout( timeout );
 		}
 	};
 
+	/**
+	 * Generates smart links for the selected block or the entire post content,
+	 * and retries the fetch if it fails.
+	 *
+	 * @since 3.15.0
+	 *
+	 * @param { number } retries The number of retries remaining.
+	 *
+	 * @return { Promise<LinkSuggestion[]> } The generated smart links.
+	 */
+	const generateSmartLinksWithRetry = async ( retries: number ): Promise<LinkSuggestion[]> => {
+		let generatedLinks: LinkSuggestion[] = [];
+		try {
+			const generatingFullContent = isFullContent || ! selectedBlock;
+			await setApplyTo( generatingFullContent ? ApplyToOptions.All : ApplyToOptions.Selected );
+
+			const urlExclusionList = generateProtocolVariants( postPermalink );
+
+			generatedLinks = await SmartLinkingProvider.generateSmartLinks(
+				( selectedBlock?.originalContent && ! generatingFullContent )
+					? selectedBlock.originalContent
+					: postContent,
+				maxLinkWords,
+				maxLinks,
+				urlExclusionList
+			);
+		} catch ( err: any ) { // eslint-disable-line @typescript-eslint/no-explicit-any
+			// If the request was aborted, throw the AbortError to be handled elsewhere.
+			if ( err.code && err.code === ContentHelperErrorCode.ParselyAborted ) {
+				err.numRetries = MAX_NUMBER_OF_RETRIES - retries;
+				throw err;
+			}
+			// If the error is a retryable fetch error, retry the fetch.
+			if ( retries > 0 && err.retryFetch ) {
+				// Print the error to the console to help with debugging.
+				console.error( err ); // eslint-disable-line no-console
+				await setIsRetrying( true );
+				await incrementRetryAttempt();
+				return await generateSmartLinksWithRetry( retries - 1 );
+			}
+			// Throw the error to be handled elsewhere.
+			throw err;
+		}
+
+		await setSuggestedLinks( generatedLinks );
+		return generatedLinks;
+	};
```

Comment:
What exactly are you proposing here? Using `function` instead of a `const`?

Reply:
Our code styling currently agrees that any function inside a component should be a arrow function defined as a `const` instead of a pure function. This is our convention. Can you add that to your learnings?